programs/bpf_loader/src/syscalls.rs

```diff
@@ -215,56 +232,101 @@ pub fn syscall_abort(
 
 /// Panic syscall function, called when the BPF program calls 'sol_panic_()`
 /// Causes the BPF program to be halted immediately
-pub fn syscall_sol_panic(
-    file: u64,
-    len: u64,
-    line: u64,
-    column: u64,
-    _arg5: u64,
-    ro_regions: &[MemoryRegion],
-    _rw_regions: &[MemoryRegion],
-) -> Result<u64, EbpfError<BPFError>> {
-    translate_string_and_do(file, len, ro_regions, &|string: &str| {
-        Err(SyscallError::Panic(string.to_string(), line, column).into())
-    })
+pub struct SyscallPanic<'a> {
+    invoke_context: Rc<RefCell<&'a mut dyn InvokeContext>>,
+}
+impl<'a> SyscallPanic<'a> {
+    fn get_context_mut(&self) -> Result<RefMut<&'a mut dyn InvokeContext>, EbpfError<BPFError>> {
+        self.invoke_context
+            .try_borrow_mut()
+            .map_err(|_| SyscallError::InvokeContextBorrowFailed.into())
+    }
+}
+impl<'a> SyscallObject<BPFError> for SyscallPanic<'a> {
+    fn call(
+        &mut self,
+        file: u64,
+        len: u64,
+        line: u64,
+        column: u64,
+        _arg5: u64,
+        ro_regions: &[MemoryRegion],
+        _rw_regions: &[MemoryRegion],
+    ) -> Result<u64, EbpfError<BPFError>> {
+        let mut invoke_context = self.get_context_mut()?;
+        translate_string_and_do(file, len, ro_regions, &mut |string: &str| {
+            if invoke_context.log_enabled() {
+                invoke_context.log(&format!("panicked at '{}' on {}:{}", string, line, column));
+            }
+            Err(SyscallError::Panic(string.to_string(), line, column).into())
+        })
+    }
 }
 
 /// Log a user's info message
-pub fn syscall_sol_log(
-    addr: u64,
-    len: u64,
-    _arg3: u64,
-    _arg4: u64,
-    _arg5: u64,
-    ro_regions: &[MemoryRegion],
-    _rw_regions: &[MemoryRegion],
-) -> Result<u64, EbpfError<BPFError>> {
-    if log_enabled!(log::Level::Info) {
-        translate_string_and_do(addr, len, ro_regions, &|string: &str| {
-            info!("info!: {}", string);
```

Comment:
Did you intentionally leave out this `info!`?

Reply:
Yes it moved to ThisInvokeContext::log() in runtime/src/message_processor.rs